x/feegrant/keeper/keeper.go

```diff
@@ -52,8 +53,12 @@ func (k Keeper) GrantAllowance(ctx sdk.Context, granter, grantee sdk.AccAddress,
 	key := feegrant.FeeAllowanceKey(granter, grantee)
 
 	var oldExp *time.Time
-	existingGrant, err := k.getGrant(ctx, grantee, granter)
-	if err != nil && existingGrant != nil && existingGrant.GetAllowance() != nil {
+	existingGrant, err := k.getGrant(ctx, granter, grantee)
+	if err != nil && !strings.Contains(err.Error(), "fee-grant not found") {
+		return err
+	}
```

Comment:
Can you explain what the bug was before?

The string.Contains in state-machine code is too fragile, someone might change the error message one day.

Reply:
updated the desc.